metal/datasource_metal_vlan.go

```diff
@@ -0,0 +1,124 @@
+package metal
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/packethost/packngo"
+)
+
+func dataSourceMetalVlan() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceMetalVlanRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"vlan_id"},
+				Description:   "ID of parent project of the VLAN. Use together with vxland",
+			},
+			"vxlan": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"vlan_id"},
+				Description:   "VXLAN numner of the VLAN. Unique in a project. Use with project_id",
```

Comment:
https://github.com/equinix/terraform-provider-metal/issues/69#issuecomment-820445819

It sounds like we can't count on these being unique.

How does this sound?

```hcl
data "metal_vlan" "vxlan1000" {
 vxlan = 1000
 project_id = "required"
 
 metro = "optional"
 facility = "optional"
 
 // error if more than 1 match
}
```

Reply:
I've just came here to point this out. I will need to rework the lookup.

Reply:
I asked about convention for this kind of optional filtering here: https://hashicorp.slack.com/archives/C0HSE5WMB/p1618495823010700

I can't think of examples of this, but I expect it will be fine.

Reply:
Seems we do this here: https://github.com/equinix/terraform-provider-metal/blob/main/metal/datasource_metal_project.go#L165